test/src/test/java/hudson/PluginManagerTest.java

```diff
@@ -533,4 +535,20 @@ public void findResourceForPluginFirstClassLoader() throws Exception {
 
         assertEquals(fromPlugin, fromToolkit);
     }
+
+    // Sources for jenkins-50336.hpi are available at https://github.com/Vlatombe/jenkins-50336
+    //
+    // package io.jenkins.plugins;
+    // import org.jenkinsci.plugins.variant.OptionalExtension;
```

Comment:
So this does not actually test the much more common case of `@Extension(optional = true)`?